### PR TITLE
[full-ci]Enforce pass by default. changed api tests

### PR DESCRIPTION
**Pull request by ScharfViktor**
related https://github.com/owncloud/ocis/issues/7682

Changed api tests. public link is password protected by default. 

Note: protected linking is disabled for the webUi and e2e tests, as changes must be made to the web first

### CODE REVIEW

**Comment by ScharfViktor** [comment]
@saw-jan could you please have look litmus test? Maybe we can temporarily skip the `public-share` test and fix it in another PR?

File: tests/acceptance/expected-failures-API-on-OCIS-storage.md [hunk 1]
```diff
@@ -587,5 +587,7 @@ Not everything needs to be implemented for ocis. While the oc10 testsuite covers
 - [coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature:22](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature#L22)
 - [coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature:23](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature#L23)
 
+#### [sharing the shares folder to users exits with different status code than in oc10 backend](https://github.com/owncloud/ocis/issues/7717)
+
```
Comment (saw-jan) [inline tests/acceptance/expected-failures-API-on-OCIS-storage.md:591]: ?

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -97,6 +97,11 @@ class FeatureContext extends BehatVariablesContext {
 	private string $publicLinkSharePassword = '';
 	private string $ocPath = '';
 
+	// /**
+	//  * The secure password to use in tests that create public link shares
+	//  */
+	// private string $securePublicLinkSharePassword = '';
+
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FeatureContext.php:104]:
```suggestion
```

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -494,7 +499,8 @@ public function __construct(
 		$this->alt4UserPassword = "ThisIsThe4thAlternatePwd";
 		$this->subAdminPassword = "IamAJuniorAdmin42";
 		$this->alternateAdminPassword = "IHave99LotsOfPriv";
-		$this->publicLinkSharePassword = "publicPwd1";
+		$this->publicLinkSharePassword = "publicPwd:1";
+		// $this->securePublicLinkSharePassword = "Pass:1234";
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FeatureContext.php:503]: can be removed now?
```suggestion
```

File: tests/acceptance/features/coreApiSharePublicLink1/accessToPublicLinkShare.feature [hunk 4]
```diff
@@ -33,23 +34,25 @@ Feature: accessing a public link share
...
   Scenario: access to the preview of public shared file inside a folder without password
-    Given user "Alice" has created folder "FOLDER"
+    Given the config "OCIS_SHARING_PUBLIC_SHARE_MUST_HAVE_PASSWORD" has been set to "false"
+    And user "Alice" has created folder "FOLDER"
     And user "Alice" has uploaded file "filesForUpload/testavatar.jpg" to "FOLDER/testavatar.jpg"
     And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "FOLDER/textfile0.txt"
     And user "Alice" has created a public link share with settings
       | path        | /FOLDER |
       | permissions | change  |
+
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/accessToPublicLinkShare.feature:55]:
```suggestion
```

File: tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature [hunk 5]
```diff
@@ -187,12 +145,13 @@ Feature: create a public link share
...
-
+  @env-config
   Scenario Outline: getting the share information of password less public-links hides credential placeholders
-    Given using OCS API version "<ocs_api_version>"
+    Given the config "OCIS_SHARING_PUBLIC_SHARE_MUST_HAVE_PASSWORD" has been set to "false"
+    And using OCS API version "<ocs_api_version>"
     And user "Alice" has uploaded file with content "Random data" to "/randomfile.txt"
     When user "Alice" creates a public link share using the sharing API with settings
-      | path | randomfile.txt |
+      | path     | randomfile.txt |
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/createPublicLinkShare.feature:154]:
```suggestion
      | path | randomfile.txt |
```

File: tests/acceptance/expected-failures-API-on-OCIS-storage.md (hunk 1 above)
Comment (ScharfViktor) [reply tests/acceptance/expected-failures-API-on-OCIS-storage.md:591]: ah, need delete it

**Review by 2403905 — APPROVED** [review]
The defaultconfig.go looks good

**Pull request merged**